### PR TITLE
[Alamin Step 7] Let's add/update/delete tasks

### PR DESCRIPTION
**Pull request by rr-mdalauddin-alamin**
### Step 7: Let's add/update/delete tasks

- Let's create a task list screen, create screen, detail screen, and edit screen
  - create controller and view with `rails generate` command
  - Let's add necessary implementations to our controllers and views
  - Display flash messages on screen after creation, update and deletion
- Edit `routes.rb` so that `http://localhost:3000/` shows the list of tasks
- Install [RuboCop](https://github.com/rubocop/rubocop) (a static code analysis tool for Ruby) or [fablicop](https://github.com/Fablic/fablicop) in `Gemfile`. prize. Check the README for settings and how to execute commands.
- Create a PR on GitHub and get it reviewed
  - Consider splitting the PR into two or more times if the PR is likely to grow in the future

### Screenshot

image

image

image

### CODE REVIEW

**Comment by Katsumi-N** [comment]
@alaminshakeel 
Could you add demo on PR description? (screenshots or movie)

File: myapp/app/controllers/task_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def new
+    @task = Task.new
+    @labels = Label.all
+  end
+
+  def create
+    @task = Task.new(task_params)
+    @task.user_id = 1
```
Comment (Katsumi-N) [inline myapp/app/controllers/task_controller.rb:13]: User function will be implemented later step, so you specified user_id?

File: myapp/app/views/task/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<%= form_with model: @task, url: form_url, method: form_method, local: true  do |form| %>
```
Comment (Katsumi-N) [inline myapp/app/views/task/_form.html.erb:1]: If label does not exist, page returns this error
スクリーンショット 2023-08-07 8 46 28

So you may add initial value in `db/seeds.rb`
(ex)
```ruby
label = Label.create(
  name: 'Test label'
)
```

The remaining `label` functions should be implemented in step20.
https://github.com/Fablic/training/blob/develop/steps_en.md#step-20-lets-be-able-to-put-labels-on-tasks

File: myapp/app/views/task/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <td><%= @task.due_date %></td>
+  </tr>
+  <tr>
+    <th>Status:</th>
+    <td><%= @task.status %></td>
+  </tr>
+</table>
+
+<%= link_to 'Edit', edit_task_path(@task) %> |
+<%= link_to 'All Taskst', task_index_path %>
```
Comment (Katsumi-N) [inline myapp/app/views/task/show.html.erb]: `Tasks`?

File: myapp/app/controllers/task_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      flash[:success] = "Task was successfully created."
+      redirect_to task_index_path(@task)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    @labels = @task.labels
```
Comment (Katsumi-N) [inline myapp/app/controllers/task_controller.rb]: You give `@labels` here, but `myapp/app/views/task/show.html.erb` does not show label.
You should choose to remove `@labels` here or show labels at `show.html.erb`

File: myapp/app/controllers/task_controller.rb (hunk 1 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/task_controller.rb:13]: Previous step-7 had "user_id" column as NOT NULL. so i needed to provide value.
so do you want me to migrate to NULL or removing user_id column?

File: myapp/app/views/task/_form.html.erb (hunk 2 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/views/task/_form.html.erb:1]: We dont have Label model implementation in this step. so its better to remove integration of "labels" for now

File: myapp/app/controllers/task_controller.rb (hunk 4 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/task_controller.rb]: Its better to remove according to this step

**Comment by rr-mdalauddin-alamin** [comment]
> @alaminshakeel Could you add demo on PR description? (screenshots or movie)

Added! please check

File: myapp/app/controllers/task_controller.rb (hunk 1 above)
Comment (Katsumi-N) [reply myapp/app/controllers/task_controller.rb:13]: I just wanted to confirm, it is ok

**Review by Katsumi-N — APPROVED** [review]
I tested on local and it worked :thumbsup:
LGTM

**Pull request merged**